src/Collection.php

```diff
@@ -890,6 +901,14 @@ public function truthy(): bool
         return (new Truthy())()($this)->current();
     }
 
+    /**
+     * @template U
+     *
+     * @param callable(U ...): list<U> $callback
+     * @param array<int, U> $parameters
+     *
+     * @return Collection<int, list<U>>
```

Comment:
```suggestion
     * @return CollectionInterface<int, list<U>>
```

shouldn't these return the interface?

Reply:
I removed the annotations, so I guess this should be considered as fixed :D